File: backend/src/main/scala/ch/countrygame/RelationNamed.scala
package ch.countrygame

import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

case class RelationNamed(relationId: Int, countryName1: String, countryName2: String, relationType: Int) {
}
object RelationNamed {
  implicit val encoder = deriveEncoder[RelationNamed]
  implicit val decoder = deriveDecoder[RelationNamed]
}

File: frontend/src/app/models/relation-named.ts
export class RelationNamed {
  relationId: number;
  countryName1: String;
  countryName2: String;
  relationType: number;

  constructor(id: number, countryName1: String, countryName2: String, relationType: number) {
    this.relationId = id;
    this.countryName1 = countryName1;
    this.countryName2 = countryName2;
    this.relationType = relationType;
  }
}
